pkg/querier/store_gateway_client.go

```diff
@@ -85,9 +92,9 @@ func newStoreGatewayClientPool(discovery client.PoolServiceDiscovery, clientConf
 		TLS:                 clientConfig.TLS,
 	}
 	poolCfg := client.PoolConfig{
-		CheckInterval:      time.Minute,
+		CheckInterval:      clientConfig.PoolConfig.CleanupPeriod,
 		HealthCheckEnabled: true,
-		HealthCheckTimeout: 10 * time.Second,
+		HealthCheckTimeout: clientConfig.PoolConfig.RemoteTimeout,
```

Comment:
Could we find better sane defaults instead of adding new config options? I'm also not super sure it's a good idea mixing the health check timeout with the connect timeout (the new `RemoteTimeout` is also used as "connect timeout" in the dial function).

Give we had a health check timeout of 10 seconds, maybe we just want to run the check every 10s too? I don't think we need to get very aggressive on detecting unhealthy store-gateways. The real value in this PR will be the timeout on the connect.

Reply:
Sounds good. I prefer not having to add config options. The main change I had in mind was to reduce `CheckInterval` from `1m`, so that queries don't just wait for 30s (on avg) before the querier realizes the store-gateway is unresponsive.

In that case I'll also just hardcode a connection timeout to 10s.